images/kcp-registrar/register.sh

```diff
@@ -185,21 +185,38 @@ switch_ws() {
     KUBECONFIG="$kcp_kcfg" kubectl kcp workspace current
 }
 
+get_sync_target_name() {
+    local cluster_name="$1"
+
+    if [[ -z ${cluster_name} ]]; then
+        exit_error "Can't return Sync Tartge Name because of missing cluster name!\n\n"
+    fi
+
+    local len=${#cluster_name}
+    if [ $len -le 43 ]; then
+        echo "$cluster_name"
+    else
+        echo "$(echo -n "$cluster_name" | md5sum | cut -d ' ' -f1)"
+    fi
+}
+
 register() {
     printf "Getting the list of registered clusters\n"
     existing_clusters=$(KUBECONFIG=${kcp_kcfg} kubectl get synctargets -o name)
 
+    local sync_target_name
     for i in "${!clusters[@]}"; do
         printf "Processing cluster %s (%s/%s)\n" "${clusters[$i]}" "$((i+1))" "${#clusters[@]}"
         if echo "${existing_clusters}" | grep "${clusters[$i]}"; then
             printf "Cluster already registered\n"
         else
             printf "Registering cluster\n"
             syncer_manifest="/tmp/syncer-${clusters[$i]}.yaml"
-            KUBECONFIG="${kcp_kcfg}" kubectl kcp workload sync "${clusters[$i]}" \
+            sync_target_name="$(get_sync_target_name ${clusters[$i]})"
+            KUBECONFIG="${kcp_kcfg}" kubectl kcp workload sync "${sync_target_name}" \
                 --syncer-image "ghcr.io/kcp-dev/kcp/syncer:$KCP_SYNC_TAG" \
                 --resources deployments.apps,services,ingresses.networking.k8s.io,pipelines.tekton.dev,pipelineruns.tekton.dev,tasks.tekton.dev,runs.tekton.dev,networkpolicies.networking.k8s.io \
-                --output-file "$syncer_manifest"
+                --output-file "$syncer_manifest" >/dev/null 2>&1
```

Comment:
Why redirecting everything (including error output) to /dev/null? This means no troubleshooting information if there is an issue.

Reply:
fixed

Reply:
As discussed, it's often useful to see what's happening to the resources (e.g. created vs updated). It will keep the important steps and resource creations in the logs. 

Fixed by removing `>/dev/null 2>&1`